### PR TITLE
feat(stdlib): add json5 module with JSON5 spec extensions

### PR DESCRIPTION
**Pull request by t0k0sh1**
## Summary

- New `json5` stdlib module mirroring `json`'s 8-function surface (`load[T]` / `dump` / `stringify` / `stringifySafe`) but accepting JSON5 extensions on input: line/block comments, trailing commas, single-quoted strings, line continuation, unquoted ASCII identifier keys, hex literals, leading/trailing decimal points, `Infinity`/`-Infinity`/`NaN`, and explicit positive sign.
- Output is strict-JSON compatible by default; non-finite floats emit JSON5 bare tokens (`"NaN"` / `"Infinity"` / `"-Infinity"`) and `stringifySafe(NaN)` returns `Ok` (the one semantic difference from `json`).
- Fixes a latent dispatcher race: both `dispatchJson` and `dispatchJson5` now gate on `native_fn_sigs_` so `from json5 import load` programs do not get hijacked by the strict-json parser.

Closes #1855

## Test plan

- [x] `./build-rust/ry_tests` — 2860/2860 PASS (37 json5 unit tests + 3 json5 codegen-reject + 3 json5 typed-coll panic)
- [x] `./build-rust/ry test -p` — 213 spec files / 0 failures (json5 spec 66/66, json spec 128/128 no regression)
- [x] `./.claude/skills/pre-commit-checklist/run-asan.sh` — exit 0 (allocate-then-fail leak regression covered for str / list / map values + unterminated block comment)
- [x] `./.claude/skills/pre-commit-checklist/run-static-analysis-all.sh` — scan-build clean on fresh build (the transient "potential leak" surfaced once was confirmed false-positive via ASan coverage of the same path)
- [x] `./.claude/skills/pre-commit-checklist/run-fuzz.sh` — fuzz_parser / fuzz_json / fuzz_utf8 / fuzz_io_open each 60s, no crashes (fuzz_json confirms no regression from the dispatcher gating change)
- [x] `./.claude/skills/pre-commit-checklist/run-prompt-refs-lint.sh` — clean

## Design notes (committed in plan)

- Implementation language: **C++** (matches existing `json.cpp`)
- API scope: **full 8-function parity** with `json`
- Parser sharing: **json5 independent** (~210 lines of duplication accepted to keep `json.cpp` untouched)

## Known deferral

- `fuzz_json5` harness is compiled and seed corpus is in place, but the docker entrypoint allowlist update only takes effect after the next `/ci-image-workflow` rebuild. `run-fuzz.sh` is annotated; re-add the docker call after the rebuild.

## Release Notes

* **New Features**
  * Added a `json5` standard library module for JSON5 parsing and serialization
  * Provides `load[T]`, `dump`, `stringify`, and `stringifySafe` (including `File` overloads)
  * Supports JSON5 syntax (comments, trailing commas, single quotes, unquoted keys, hex, multiline continuations, and non-finite float literals)
  * Includes `fuzz_json5` fuzz target with a seed corpus

* **Bug Fixes**
  * Prevented JSON5 `load[T]` from being incorrectly routed to the strict JSON implementation

* **Documentation**
  * Added JSON5 reference docs and updated module import constraints for `json5`

* **Tests**
  * Added runtime and codegen coverage for JSON5 behavior and edge cases

### CODE REVIEW

File: docs/reference/json5.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+```ry
+from json5 import load
+
+src = "{\n  // tool config\n  name: 'ry',\n  port: 0xFF,\n  ratio: .5,\n}"
+case load[Map<str, any>](src):
+  Ok(cfg):
+    print(stringify(cfg, sortKeys=true))
+  Err(e):
+    print("parse error: " + e.message)
+```
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply docs/reference/json5.md:100]: Applied — `stringify` added to the import statement in 6dc1ae9 (next push). Thanks!

File: src/runtime/native/json5.cpp [hunk 2]
```diff
@@ -0,0 +1,1044 @@
...
+            // error).
+            if (!at_end() && is_id_part(src[pos])) return fail("invalid hex literal at ");
+            uint64_t v = 0;
+            for (size_t i = hex_start; i < pos; i++) {
+                v = (v << 4) | (uint64_t)hex_digit_value(src[i]);
+            }
+            int64_t signed_v = negative ? -(int64_t)v : (int64_t)v;
+            out = anyFromInt(signed_v);
+            return true;
+        }
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply src/runtime/native/json5.cpp:541]: Applied. The fix:

- Rejects hex literals exceeding 16 digits (>64 bits) with "hex literal overflow at <pos>" — matches the decimal path's `strtoll` ERANGE check
- For 16-digit hex that exceeds INT64_MAX, also rejects (previously silently reinterpreted as int64)
- Special-cases `-0x8000000000000000` to produce `INT64_MIN` without invoking `-(int64_t)INT64_MIN` UB
- Spec tests added for both the 17-digit overflow case and the INT64_MIN edge

File: tests/test_codegen_call_json5_reject.cpp [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        close(pipeOut[0]);
+        close(pipeOut[1]);
+        return {};
+    }
+
+    std::vector<const char *> argv{RY_BINARY_PATH};
+    argv.insert(argv.end(), args.begin(), args.end());
+    argv.push_back(nullptr);
+    setenv("RY_ENV", "internal", 1);
+
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply tests/test_codegen_call_json5_reject.cpp:49]: Applied. The fix saves/restores the parent's `RY_ENV` so the value does not leak to subsequent tests.

The `setenv` call stays in the parent (rather than moving it to the child between fork and execv) because `setenv` is not async-signal-safe — calling it in the child can deadlock on libc / allocator locks if another thread in the test runner is holding them. A comment documenting the trade-off has been added next to the call.
